shared/board.ts

```diff
@@ -0,0 +1,37 @@
+import { Team } from './team';
+
+export class Board {
+  protected spaces: Team[] = new Array(this.rows * this.cols).fill(Team.Empty);
+
+  constructor(private rows: number, private cols: number) {}
+
+  get squares() {
+    return [...this.spaces];
```

Comment:
what does this do? I know it's in the original, but I forgot.

Reply:
it just makes sure that anything accessing the data from the outside doesn't accidentally manipulate the array